design/definitions.go

```diff
@@ -652,9 +664,12 @@ func (r *ResourceDefinition) DSL() func() {
 }
 
 // Finalize is run post DSL execution. It merges response definitions, creates implicit action
-// parameters, initializes querystring parameters and sets path parameters as non zero attributes.
+// parameters, initializes querystring parameters, sets path parameters as non zero attributes
+// and sets the fallbacks for security methods.
 func (r *ResourceDefinition) Finalize() {
 	r.IterateActions(func(a *ActionDefinition) error {
+		a.Finalize()
```

Comment:
The code cannot do that (call `Finalize` on other definitions) otherwise we risk infinite loops. Instead there should be an additional "DefinitionSet" for the security definitions that is returned last to the DSL engine when iterating the design definition sets.